src/internal/m365/onedrive/permission.go

```diff
@@ -82,6 +83,56 @@ func getCollectionMetadata(
 	return meta, nil
 }
 
+// Unlike permissions, link shares inherit them from all the parents
+// of an item and not just the direct parent.
+func computePreviousLinkShares(
+	ctx context.Context,
+	originDir path.Path,
+	parentMetas map[string]metadata.Metadata,
+) ([]metadata.LinkShare, error) {
+	var (
+		parent     path.Path
+		linkShares []metadata.LinkShare
+	)
+
+	parent = originDir
+
+	for {
+		newParent, err := parent.Dir()
+		if err != nil {
+			return nil, clues.New("getting parent").WithClues(ctx)
+		}
+
+		if parent == newParent {
+			break
+		}
```

Comment:
Minor: this method of handling the loop doesn't make it very clear what the state and conditions are.  I'd recommend the following instead:

```suggestion
	parent = originDir.Dir()

	for len(parent.Elements()) > 0 {
	
	// ... down at the bottom  of the loop
	parent, err = parent.Dir()
	if err != nil {
		return nil, clues.New("getting parent").WithClues(ctx)
	}
```